rebac-admin-backend/v1/identities.go

```diff
@@ -0,0 +1,176 @@
+	identity := new(r.Identity)
+	defer req.Body.Close()
+
+	err := json.NewDecoder(req.Body).Decode(identity)
+	if err != nil {
+		writeErrorResponse(w, err)
+		return
+	}
+
+	identity, err = h.Identities.CreateIdentity(identity)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 201, identity)
+}
+
+// DeleteIdentitiesItem Remove an identity.
+// (DELETE /identities/{id})
+func (h handler) DeleteIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	_, err := h.Identities.DeleteIdentity(id)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	w.WriteHeader(200)
+}
+
+// GetIdentitiesItem Get a single identity.
+// (GET /identities/{id})
+func (h handler) GetIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	identity, err := h.Identities.GetIdentity(id)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 200, identity)
+}
+
+// PutIdentitiesItem Update an identity.
+// (PUT /identities/{id})
+func (h handler) PutIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	identity := new(r.Identity)
+	defer req.Body.Close()
+
+	err := json.NewDecoder(req.Body).Decode(identity)
+	if err != nil {
+		writeErrorResponse(w, err)
+		return
+	}
+
+	if id != *identity.Id {
+		writeErrorResponse(w, NewValidationError("Identity ID from path does not match the Identity object"))
+		return
+	}
+
+	identity, err = h.Identities.UpdateIdentity(identity)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 200, identity)
+}
+
+// GetIdentitiesItemEntitlements List entitlements the identity has.
+// (GET /identities/{id}/entitlements)
+func (h handler) GetIdentitiesItemEntitlements(w http.ResponseWriter, req *http.Request, id string, params r.GetIdentitiesItemEntitlementsParams) {
+	entitlements, err := h.Identities.GetIdentityEntitlements(id, &params)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := r.GetIdentityEntitlementsResponse{
+		Data:   entitlements.Data,
+		Status: 200,
+	}
+
+	writeResponse(w, 200, response)
+}
+
+// PatchIdentitiesItemEntitlements Add or remove entitlement to/from an identity.
+// (PATCH /identities/{id}/entitlements)
+func (h handler) PatchIdentitiesItemEntitlements(w http.ResponseWriter, req *http.Request, id string) {
+	w.WriteHeader(http.StatusNotImplemented)
+}
+
+// GetIdentitiesItemGroups List groups the identity is a member of.
+// (GET /identities/{id}/groups)
+func (h handler) GetIdentitiesItemGroups(w http.ResponseWriter, req *http.Request, id string, params r.GetIdentitiesItemGroupsParams) {
```

Comment:
// GetGroups returns the list of groups the identity is a member of.

we can keep the method names simple because we know this handler deals with identities. I would rather not put the http method into the method name.

Reply:
see answer below